src/analysisd/alerts/exec.c

```diff
@@ -106,46 +107,145 @@ void OS_Exec(int execq, int *arq, const Eventinfo *lf, const active_response *ar
+                        snprintf(exec_msg, OS_SIZE_1024,
+                                "%s %c%c%c %s %s %s %s %ld.%ld %d %s %s %s",
+                                lf->location,
+                                (ar->location & ALL_AGENTS) ? ALL_AGENTS_C : NONE_C,
+                                (ar->location & REMOTE_AGENT) ? REMOTE_AGENT_C : NONE_C,
+                                (ar->location & SPECIFIC_AGENT) ? SPECIFIC_AGENT_C : NONE_C,
+                                ar->agent_id != NULL ? ar->agent_id : "(null)",
+                                ar->name,
+                                user,
+                                ip,
+                                (long int)lf->time.tv_sec,
+                                __crt_ftell,
+                                lf->generated_rule->sigid,
+                                lf->location,
+                                filename ? filename : "-",
+                                extra_args ? extra_args : "-");
+                    } else {
+                        snprintf(exec_msg, OS_SIZE_1024,
+                                "(local_source) %s %c%c%c %s %s %s %s %ld.%ld %d %s %s %s",
+                                lf->location,
+                                (ar->location & ALL_AGENTS) ? ALL_AGENTS_C : NONE_C,
+                                (ar->location & REMOTE_AGENT) ? REMOTE_AGENT_C : NONE_C,
+                                (ar->location & SPECIFIC_AGENT) ? SPECIFIC_AGENT_C : NONE_C,
+                                ar->agent_id != NULL ? ar->agent_id : "(null)",
+                                ar->name,
+                                user,
+                                ip,
+                                (long int)lf->time.tv_sec,
+                                __crt_ftell,
+                                lf->generated_rule->sigid,
+                                lf->location,
+                                filename ? filename : "-",
+                                extra_args ? extra_args : "-");
+                    }
+                }
+
+            }
+            
         } else {
-            snprintf(exec_msg, OS_SIZE_1024,
-                     "(local_source) %s %c%c%c %s %s %s %s %ld.%ld %d %s %s %s",
-                     lf->location,
-                     (ar->location & ALL_AGENTS) ? ALL_AGENTS_C : NONE_C,
-                     (ar->location & REMOTE_AGENT) ? REMOTE_AGENT_C : NONE_C,
-                     (ar->location & SPECIFIC_AGENT) ? SPECIFIC_AGENT_C : NONE_C,
-                     ar->agent_id != NULL ? ar->agent_id : "(null)",
-                     ar->name,
-                     user,
-                     ip,
-                     (long int)lf->time.tv_sec,
-                     __crt_ftell,
-                     lf->generated_rule->sigid,
-                     lf->location,
-                     filename ? filename : "-",
-                     extra_args ? extra_args : "-");
+            agent_info *agt_info = NULL;
```

Comment:
Most of this code is repeated and have the same problems that I mentioned above.
Also, you don't differentiate between specific agents and remote agents:
- Specific: The ID of the agent is configured in the AR: `ar->agent_id`.
- Remote: The ID of the agent is the one that generated the alert. You can get it using `lf->location`.

Reply:
Also, could you find and implement an easy way to not repeat the old code twice (`snprintf` functions)?

Reply:
I couldn't find the changes for the first comment.